src/main/java/org/opensrp/web/rest/ClientFormResource.java

```diff
@@ -181,10 +192,46 @@ private ClientFormMetadata getMostRecentVersion(@NonNull String formIdentifier)
         return new ResponseEntity<>(HttpStatus.CREATED);
     }
 
+    @VisibleForTesting
+    @Nullable
+    protected String checkValidJsonYamlPropertiesStructure(@NonNull String fileContentString, @NonNull String contentType) {
+        if (contentType.equals(ContentType.APPLICATION_JSON.getMimeType())) {
+            try {
+                new JSONObject(fileContentString);
+                return null;
+            } catch (JSONException ex) {
+                logger.error("JSON File upload is invalid JSON", ex);
+                return ex.getMessage();
+            }
+        } else if (isYamlContentType(contentType)) {
+            try {
+                (new MVELRuleFactory(new YamlRuleDefinitionReader())).createRule(new BufferedReader(new StringReader(fileContentString)));
+            } catch (Exception ex) {
+                logger.error("Rules file upload is invalid YAML rules file", ex);
+                return ex.getMessage();
+            }
+        } else {
+            // This is a properties file
+            try {
+                Properties properties = new Properties();
+                properties.load(new StringReader(fileContentString));
+            } catch (Exception e) {
+                e.printStackTrace();
+                return e.getMessage();
+            }
+        }
```

Comment:
Do we want this to be the default? Could we have something like the file type uploaded is not supported? Are we checking for this somewhere else?

Reply:
Yes, this method is called after the file type checks, which will only have allowed JSON, YAML & Properties file.

This check is to check if the content is syntactically correct